ostd/src/mm/frame/frame_ref.rs

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use core::{marker::PhantomData, mem::ManuallyDrop, ops::Deref, ptr::NonNull};
+
+use super::{
+    meta::{AnyFrameMeta, MetaSlot},
+    Frame,
+};
+use crate::{mm::Paddr, sync::non_null::NonNullPtr};
+
+/// A struct that can work as `&'a Frame<M>`.
+#[derive(Debug)]
+pub struct FrameRef<'a, M: AnyFrameMeta + ?Sized> {
+    inner: ManuallyDrop<Frame<M>>,
+    _marker: PhantomData<&'a Frame<M>>,
+}
+
+impl<M: AnyFrameMeta + ?Sized> FrameRef<'_, M> {
+    /// Gets a [`FrameRef`] from a raw physical address.
```

Comment:
```suggestion
    /// Borrows the [`Frame`] at the physical address as a `FrameRef`.
```

Better to make the word "borrow" appear in the doc comments. And I noticed now there is no word "from" in the doc comments, maybe rename the method to `borrow_paddr`?